File: java/src/main/java/com/github/zipcodewilmington/AlphabetValue.java
package com.github.zipcodewilmington;

import java.util.Scanner;

/**
 * Marc McGough
 * Date: 11/03/2025
 *
 * AlphabetValue - Assigns numerical values to letters (a=1, b=2, etc.)
 */
public class AlphabetValue {

    /**
     * Returns the numerical value of a letter (a=1, b=2, c=3, etc.)
     * @param letter the letter to get the value for
     * @return the numerical value (1-26), or 0 if not a valid letter
     */
    public static int getLetterValue(char letter) {
        letter = Character.toLowerCase(letter);
        if (letter >= 'a' && letter <= 'z') {
            return letter - 'a' + 1;
        }
        return 0;
    }

    /**
     * Calculates the sum of all letter values in a word
     * @param word the word to calculate the value for
     * @return the sum of all letter values
     */
    public static int getWordValue(String word) {
        int total = 0;
        for (char c : word.toCharArray()) {
            total += getLetterValue(c);
        }
        return total;
    }

    /**
     * Displays the alphabet with corresponding numerical values
     */
    public static void displayAlphabet() {
        System.out.println("\nAlphabet Numerical Values:");
        System.out.println("==========================");
        for (char c = 'a'; c <= 'z'; c++) {
            System.out.printf("%c = %d%n", c, getLetterValue(c));
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Welcome to Alphabet Value Calculator!");

        displayAlphabet();

        System.out.println("\nEnter a letter or word to get its numerical value (or 'quit' to exit):");

        while (true) {
            System.out.print("\n> ");
            String input = scanner.nextLine().trim();

            if (input.equalsIgnoreCase("quit")) {
                System.out.println("Goodbye!");
                break;
            }

            if (input.isEmpty()) {
                continue;
            }

            if (input.length() == 1) {
                char letter = input.charAt(0);
                int value = getLetterValue(letter);
                if (value > 0) {
                    System.out.println("Letter '" + letter + "' has a value of: " + value);
                } else {
                    System.out.println("Invalid letter. Please enter a letter from a-z.");
                }
            } else {
                int wordValue = getWordValue(input);
                System.out.println("Word '" + input + "' has a total value of: " + wordValue);

                // Show individual letter values
                System.out.print("Breakdown: ");
                for (int i = 0; i < input.length(); i++) {
                    char c = input.charAt(i);
                    int value = getLetterValue(c);
                    if (value > 0) {
                        System.out.print(c + "=" + value);
                        if (i < input.length() - 1) {
                            System.out.print(" + ");
                        }
                    }
                }
                System.out.println();
            }
        }

        scanner.close();
    }
}

File: python/alphabet_value.py
"""
Alphabet Value Calculator - Assigns numerical values to letters (a=1, b=2, etc.)

@author Marc McGough
@date 11/03/2025
"""


def get_letter_value(letter):
    """
    Returns the numerical value of a letter (a=1, b=2, c=3, etc.)

    Args:
        letter: the letter to get the value for

    Returns:
        int: the numerical value (1-26), or 0 if not a valid letter
    """
    letter = letter.lower()
    if 'a' <= letter <= 'z':
        return ord(letter) - ord('a') + 1
    return 0


def get_word_value(word):
    """
    Calculates the sum of all letter values in a word

    Args:
        word: the word to calculate the value for

    Returns:
        int: the sum of all letter values
    """
    total = 0
    for char in word:
        total += get_letter_value(char)
    return total


def display_alphabet():
    """Displays the alphabet with corresponding numerical values"""
    print("\nAlphabet Numerical Values:")
    print("==========================")
    for i in range(26):
        char = chr(ord('a') + i)
        print(f"{char} = {get_letter_value(char)}")


def main():
    """Main function to run the alphabet value calculator"""
    print("Welcome to Alphabet Value Calculator!")

    display_alphabet()

    print("\nEnter a letter or word to get its numerical value (or 'quit' to exit):")

    while True:
        user_input = input("\n> ").strip()

        if user_input.lower() == 'quit':
            print("Goodbye!")
            break

        if not user_input:
            continue

        if len(user_input) == 1:
            value = get_letter_value(user_input)
            if value > 0:
                print(f"Letter '{user_input}' has a value of: {value}")
            else:
                print("Invalid letter. Please enter a letter from a-z.")
        else:
            word_value = get_word_value(user_input)
            print(f"Word '{user_input}' has a total value of: {word_value}")

            # Show individual letter values
            breakdown = []
            for char in user_input:
                value = get_letter_value(char)
                if value > 0:
                    breakdown.append(f"{char}={value}")

            if breakdown:
                print(f"Breakdown: {' + '.join(breakdown)}")


if __name__ == "__main__":
    main()
